src/modules/application/application.service.spec.ts

```diff
@@ -1,19 +1,127 @@
+} from '@nestjs/typeorm';
+import { Application } from '../application/application.entity';
+import * as TestDbCOnfig from '../../../test/config';
+import { Connection, EntityManager, QueryRunner, Repository } from 'typeorm';
+import { Chance } from 'chance';
+import { Logger } from '../logger/logger.service';
 import { ApplicationService } from './application.service';
+import { OrganizationService } from '../organization/organization.service';
+import { organizationFixture } from '../organization/organization.fixture';
+import { Organization } from '../organization/organization.entity';
+import { applicationFixture } from './application.fixture';
 
-// TODO: fix test so pending can be removed
-xdescribe('ApplicationService', () => {
+const chance = new Chance();
+
+const MockLogger = {
+  log: jest.fn(),
+  error: jest.fn(),
+  setContext: jest.fn(),
+  info: jest.fn(),
+  debug: jest.fn(),
+};
+
+const MockOrgService = {
+  getByNamespace: jest.fn(),
+};
+
+describe('ApplicationService', () => {
   let service: ApplicationService;
+  let module: TestingModule;
+  let repo: Repository<Application>;
+  let orgRepo: Repository<Organization>;
+  let organizations: Organization[];
+  let applications: Application[];
+  let queryRunner: QueryRunner;
 
   beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
-      providers: [ApplicationService],
+    jest.resetAllMocks();
+    module = await Test.createTestingModule({
+      imports: [
+        TypeOrmModule.forRoot(TestDbCOnfig.default as TypeOrmModuleOptions),
+        TypeOrmModule.forFeature([Organization, Application]),
+      ],
+      providers: [
+        ApplicationService,
+        { provide: OrganizationService, useValue: MockOrgService },
+        {
+          provide: Logger,
+          useValue: MockLogger,
+        },
+      ],
     }).compile();
 
+    const dbConnection = module.get(Connection);
+    const manager = module.get(EntityManager);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    queryRunner = manager.queryRunner = dbConnection.createQueryRunner(
+      'master',
+    );
+    await queryRunner.startTransaction();
+    repo = module.get<Repository<Application>>(getRepositoryToken(Application));
+    orgRepo = module.get<Repository<Organization>>(
+      getRepositoryToken(Organization),
+    );
+
+    organizations = await organizationFixture(orgRepo);
+    applications = await applicationFixture(repo, organizations[0], 2);
+
     service = module.get<ApplicationService>(ApplicationService);
   });
 
-  it('should be defined', () => {
-    expect(service).toBeDefined();
+  afterEach(async () => {
+    await queryRunner.rollbackTransaction();
+    module.close();
+  });
+
+  it('create() it should not save an application with existing namespace and parentOrg ', async () => {
+    const testApp = chance.pickone(applications);
+    const organization = chance.pickone(organizations);
```

Comment:
Why the use of `chance` here? I tend to think tests should not have randomness so that they are repeatable, but maybe I don't understand the rationale here

Reply:
Yeah I agree with you that tests should not have randomness but in this case, using chance to pick one of an already created app or org doesn't affect the outcome of the tests because what is selected is what is going to be saved at the end and that is what we are asserting for.